python/ray/serve/tests/test_standalone2.py

```diff
@@ -701,5 +702,31 @@ def ready(self):
     )
 
 
+def test_handle_early_detect_failure(shutdown_ray):
```

Comment:
please add a header comment describing the behavior that's being tested (let's try to do this in general, really helps readers in the future)